transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -365,7 +388,15 @@ public ChannelFuture writeOneInbound(Object msg) {
      */
     public ChannelFuture writeOneInbound(Object msg, ChannelPromise promise) {
         if (checkOpen(true)) {
-            pipeline().fireChannelRead(msg);
+            if (!config().isAutoRead()) {
+                Queue<AbstractMap.SimpleEntry<Object, ChannelPromise>> tempInboundMessages = tempInboundMessages();
+                if (tempInboundMessages != null) {
+                    tempInboundMessages.add(new AbstractMap.SimpleEntry<Object, ChannelPromise>(msg, promise));
+                }
+                return promise;
```

Comment:
@normanmaurer 
I also add a unit test to test `ChannelPromise` with `channel.autoRead = false`.
I found a bug here. We should return promise when `channel.autoRead = false`. Otherwise, `checkException` will be executed and the `ChannelPromise` will be completed.